code/datums/datumvars.dm

```diff
@@ -748,6 +748,28 @@
 			src.give_disease(M)
 			href_list["datumrefresh"] = href_list["give_spell"]
 
+		else if(href_list["ninja"])
+			if(!check_rights(R_FUN))
+				return
+
+			var/mob/M = locate(href_list["ninja"])

```

Comment:
this locate needs to have a scope, you can use either `mob_list` or typecast it to a human right away, then you can use `carbon_list`.